sierra2mlir/src/statements/inline_jumps.rs

```diff
@@ -180,4 +180,119 @@ impl<'ctx> Compiler<'ctx> {
 
         Ok(())
     }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn inline_array_get(
+        &self,
+        id: &str,
+        statement_idx: usize,
+        region: &Region,
+        block: &Block,
+        blocks: &BTreeMap<usize, BlockInfo>,
+        invocation: &Invocation,
+        variables: &HashMap<u64, Variable>,
+        storage: &Storage,
+    ) -> Result<()> {
+        let libfunc = storage.libfuncs.get(id).unwrap();
+        let array_arg = &libfunc.get_args()[0];
+        let index_arg = &libfunc.get_args()[1];
+
+        // fallthrough if ok
+        // jump if panic
+
+        // 0 = ok block, 1 = panic block
+        let target_blocks = invocation
+            .branches
+            .iter()
+            .map(|branch| match branch.target {
+                GenBranchTarget::Fallthrough => statement_idx + 1,
+                GenBranchTarget::Statement(idx) => idx.0,
+            })
+            .map(|idx| {
+                let target_block_info = blocks.get(&idx).unwrap();
+                target_block_info
+            })
+            .collect_vec();
+
+        let target_block_info = target_blocks[0];
+        let panic_block_info = target_blocks[1];
+
+        if let SierraType::Array { ty: _, len_type, element_type } = &array_arg.ty {
+            // arg 0 is range check, can ignore
+            // arg 1 is the array
+            // arg 2 is the index
+
+            let array_var = variables
+                .get(&invocation.args[array_arg.loc].id)
+                .expect("variable array should exist");
+            let index_var = variables
+                .get(&invocation.args[index_arg.loc].id)
+                .expect("variable index should exist");
+
+            let array_value = array_var.get_value();
+
+            // get the current length
+            let length_op = self.op_llvm_extractvalue(block, 0, array_value, *len_type)?;
+            let length: Value = length_op.result(0)?.into();
+
+            // check if index is out of bounds
+            let cmp_op = self.op_cmp(block, CmpOp::UnsignedLess, index_var.get_value(), length);
+            let cmp = cmp_op.result(0)?.into();
+
+            let block_get_idx = region.append_block(Block::new(&[]));
+
+            // collect args to the panic block
+            let mut args_to_panic_block = vec![];
```

Comment:
No need for mutability here if this is done with a single map expression over `panic_block_info.variables_at_start.keys()` rather than an explicit for loop